perfmetrics/scripts/run_e2e_tests.sh

```diff
@@ -53,28 +53,162 @@ sudo apt-get install -y gcc python3-dev python3-setuptools
+  echo $BUCKET_NAME
+}
+
+# Non parallel execution of integration tests located within specified test directories for group 1.
+function run_non_parallel_tests_group_1() {
+  for test_dir_np_group_1 in "${test_dir_non_parallel_group_1[@]}"
+  do
+    test_path_non_parallel_group_1="./tools/integration_tests/$test_dir_np_group_1"
+    # Executing integration tests
+    GODEBUG=asyncpreemptoff=1 go test $test_path_non_parallel_group_1 -p 1 --integrationTest -v --testbucket=$BUCKET_NAME_NON_PARALLEL_GROUP_1 --testInstalledPackage=$run_e2e_tests_on_package -timeout $INTEGRATION_TEST_TIMEOUT
+    exit_code_non_parallel_group_1=$?
+    if [ $exit_code_non_parallel_group_1 != 0 ]; then
+      test_fail_np_group_1=$exit_code_non_parallel_group_1
+      echo "test fail in non parallel in group 1: " $test_fail_np_group_1
+    fi
+  done
+  return $test_fail_np_group_1
+}
+
+# Non parallel execution of integration tests located within specified test directories for group 2.
+function run_non_parallel_tests_group_2() {
+  for test_dir_np_group_2 in "${test_dir_non_parallel_group_2[@]}"
+  do
+    test_path_non_parallel_group_2="./tools/integration_tests/$test_dir_np_group_2"
+    # Executing integration tests
+    GODEBUG=asyncpreemptoff=1 go test $test_path_non_parallel_group_2 -p 1 --integrationTest -v --testbucket=$BUCKET_NAME_NON_PARALLEL_GROUP_2 --testInstalledPackage=$run_e2e_tests_on_package -timeout $INTEGRATION_TEST_TIMEOUT
+    exit_code_non_parallel_group_2=$?
+    if [ $exit_code_non_parallel_group_2 != 0 ]; then
+      test_fail_np_group_2=$exit_code_non_parallel_group_2
+      echo "test fail in non parallel in group 2: " $test_fail_np_group_2
+    fi
+  done
+  return $test_fail_np_group_2
+}
+
+# Parallel execution of integration tests located within specified test directories.
+# It aims to improve testing speed by running tests concurrently, while providing basic error reporting.
+function run_parallel_tests() {
+  for test_dir_p in "${test_dir_parallel[@]}"
+  do
+    test_path_parallel="./tools/integration_tests/$test_dir_p"
+    # Executing integration tests
+    GODEBUG=asyncpreemptoff=1 go test $test_path_parallel -p 1 --integrationTest -v --testbucket=$BUCKET_NAME_PARALLEL --testInstalledPackage=$run_e2e_tests_on_package -timeout $INTEGRATION_TEST_TIMEOUT &
+    pid=$!  # Store the PID of the background process
+    pids+=("$pid")  # Optionally add the PID to an array for later
+  done
+
+  # Wait for processes and collect exit codes
+  for pid in "${pids[@]}"; do
+    wait $pid
+    exit_code_parallel=$?
+    if [ $exit_code_parallel != 0 ]; then
+      test_fail_p=$exit_code_parallel
+      echo "test fail in parallel: " $test_fail_p
+    fi
+  done
+  return $test_fail_p
+}
+
+# Test setup
+# Create Bucket for non parallel e2e tests
+# The bucket prefix for the random string
+bucketPrefix="gcsfuse-non-parallel-e2e-tests-group-1-"
+BUCKET_NAME=$(create_bucket $bucketPrefix)
```

Comment:
we can directly assign to BUCKET_NAME_NON_PARALLEL_GROUP_1?

Reply:
@ankitaluthra1 added this change in her PR.